core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/projection/ProjectionOpDesc.scala

```diff
@@ -0,0 +1,40 @@
+package edu.uci.ics.texera.workflow.operators.projection
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonPropertyDescription}
+import com.google.common.base.Preconditions
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaTitle
+import edu.uci.ics.texera.workflow.common.metadata._
+import edu.uci.ics.texera.workflow.common.metadata.annotations.AutofillAttributeNameList
+import edu.uci.ics.texera.workflow.common.operators.OneToOneOpExecConfig
+import edu.uci.ics.texera.workflow.common.operators.map.MapOpDesc
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, Schema}
+
+class ProjectionOpDesc extends MapOpDesc {
+
+  @JsonProperty(required = true)
+  @JsonSchemaTitle("Attributes")
+  @JsonPropertyDescription("a subset of column to keeps")
+  @AutofillAttributeNameList
+  val attributes: List[String] = List[String]()
+
+  override def operatorExecutor: OneToOneOpExecConfig = {
+    new OneToOneOpExecConfig(operatorIdentifier, _ => new ProjectionOpExec(this))
```

Comment:
Can we only pass the `attributes` variable to the OpExec instead of `this`?

Reply:
I used to just pass `OpDesc` into `OpExec` because it's easier to write. However, `OpDesc` might carry over some more unintentional info, so I think it's good to just pass all the primitive values, although it makes coding more verbose.

Reply:
I will make the change then.